src/hotspot/share/gc/parallel/psParallelCompact.cpp

```diff
@@ -801,24 +801,38 @@ HeapWord* ParallelCompactData::calc_new_pointer(HeapWord* addr, ParCompactionMan
     return result;
   }
 
-  // Otherwise, the new location is region->destination + block offset + the
-  // number of live words in the Block that are (a) to the left of addr and (b)
-  // due to objects that start in the Block.
+  // Otherwise, the new location is region->destination + #live words to left
+  // of addr in this region. Calculating #live words naively means walking the
+  // mark bitmap from the start of this region. Block table can be used to
+  // speed up this process, but it would incur some side-effect. In debug-only
+  // code, we prefer the side-effect free version so that no side-effect will
+  // not leak into release code.
+  if (use_block_table) {
```

Comment:
I'm having trouble making sense of the last sentence of this comment, even though I think I know what is intended.  Perhaps something like "IIn debug-only code (such as asserts), we prefer the slower but side-effect free version, to avoid side effects that would not occur for release code and could potentially affect future calculations."

Reply:
Fixed; I just realized that the last sentence uses double negative, but it shouldn't... Thank you for the suggestion.